buildSrc/src/main/java/org/elasticsearch/gradle/LoggedExec.java

```diff
@@ -41,4 +46,50 @@ public LoggedExec() {
             );
         }
     }
+
+    public static ExecResult exec(Project project, Action<ExecSpec> action) {
```

Comment:
This can be deduplicated by having a generic private method with the impl like this:

```
public static ExecResult exec(Project project, Action<ExecSpec> action) {
    return genericExec(project, project::exec, action);
}


public static ExecResult javaexec(Project project, Action<JavaExecSpec> action) {
    return genericExec(project, project::javaexec, action);
}

private static <T extends BaseExecSpec>
ExecResult genericExec(Project project, Function<Action<T>,ExecResult> function, Action<T> action) {
    if (project.getLogger().isInfoEnabled()) {
        return function.apply(action);
    }
    ByteArrayOutputStream output = new ByteArrayOutputStream();
    try {
        return function.apply(spec -> {
            spec.setStandardOutput(output);
            spec.setErrorOutput(output);
            action.execute(spec);
        });
    } catch (Exception e) {
        try {
            for (String line : output.toString("UTF-8").split("\\R")) {
                project.getLogger().error(line);
            }
        } catch (UnsupportedEncodingException ue) {
            throw new GradleException("Failed to read exec output", ue);
        }
        throw e;
    }
}
```